compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -675,6 +675,26 @@ static TR::Register *shiftHelper(TR::Node *node, TR::ARM64ShiftCode shiftType, T
 TR::Register *
 OMR::ARM64::TreeEvaluator::ishlEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
+   if (node->getOpCodeValue() == TR::lshl)
+      {
+      TR::Node *firstChild = node->getFirstChild();
+      TR::Node *secondChild = node->getSecondChild();
+      if (firstChild->getOpCodeValue() == TR::i2l && secondChild->getOpCodeValue() == TR::iconst)
+         {
+         // Typical IL sequence in array access (sign-extend index value and shift it to left)
+         TR::Node *indexChild = firstChild->getFirstChild();
+         TR::Register *srcReg = cg->evaluate(indexChild);
+         TR::Register *trgReg = cg->allocateRegister();
```

Comment:
If this is the last reference to `indexChild` , can its register be re-used rather than allocating a target register?

Reply:
I changed the code to reuse `srcReg` when the reference count of `indexChild` is 1.